pkg/codegen/go/gen.go

```diff
@@ -512,10 +512,19 @@ func (pkg *pkgContext) genOutputTypes(w io.Writer, t *schema.ObjectType, details
 
 		for _, p := range t.Properties {
 			printComment(w, p.Comment, false)
-			outputType, applyType := pkg.outputType(p.Type, !p.IsRequired), pkg.plainType(p.Type, !p.IsRequired)
+			outputType, applyType := pkg.outputType(p.Type, true), pkg.plainType(p.Type, true)
```

Comment:
The API looks good, but `p.Type` must be recorded as requiring a pointer output type or we may generate references to missing types. This is tracked by `typeDetails.ptrElement`, which is set here: https://github.com/pulumi/pulumi/blob/master/pkg/codegen/go/gen.go#L1081-L1085. That code should be rewritten to recurse when it sets `ptrElement`.

Reply:
Indeed - that was indeed leading to references to types that didn't exist - good catch!